docs/02-app/01-building-your-application/06-optimizing/02-videos.mdx

```diff
@@ -0,0 +1,281 @@
+
+### Displaying Externally Hosted Videos in Next.js
+
+Efficiently displaying videos from platforms like YouTube or Vimeo in Next.js involves combining appropriate embedding techniques with dynamic loading strategies. Using server components with a loading state managed by React Suspense optimizes page load times and enhances user experience. This ensures videos are loaded responsively, adapting to user interactions and network conditions.
+
+1. **Create a Server Component for Video Embedding**
+
+   The first step is to create a [Server Component](https://nextjs.org/docs/app/building-your-application/rendering/server-components) that generates the appropriate iframe for embedding the video. This component will fetch the source URL for the video and render the iframe.
+
+   ```jsx filename="app/ui/video-component.jsx"
+   export const async function VideoComponent(){
+     const src = await getVideoSrc()
+
+     return <iframe src={src} frameborder="0" allowfullscreen />
+   }
+   ```
+
+2. **Dynamically Load the Video Component Using React Suspense**
+
+   After creating the server component for video embedding, the next step is to [dynamically load](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming) this component. [React Suspense](https://react.dev/reference/react/Suspense) is used to manage the loading fallback, showing a fallback while the video is being fetched.
+
+   ```jsx filename="app/page.jsx"
+   import { Suspense } from 'react'
+   import { VideoComponent } from '../ui/VideoComponent.server'
+
+   export default function Page() {
+     return (
+       <section>
+         <Suspense fallback={<p>Loading video...</p>}>
+           <VideoComponent />
+         </Suspense>
+         {/* Other content of the page */}
+       </section>
+     )
+   }
+   ```
+
+> **Good to know**: When embedding videos from external platforms, consider the following best practices:
+>
+> - **Responsive Embedding**: Ensure the video embeds are responsive. Use CSS to make the iframe or video player adapt to different screen sizes.
+> - **Consider Data Usage**: Implement strategies for loading videos based on network conditions, especially for users with limited data plans.
+> - **Accessibility Considerations**: Provide captions or alternative content for accessibility purposes.
+
+By integrating server components with React Suspense for loading, Next.js applications can efficiently handle externally hosted videos. This approach results in quicker load times and a smoother user experience. Such integration boosts both performance and engagement, ensuring an optimal video viewing experience on your site.
```

Comment:
```suggestion
This approach results in a btter user experience as it prevents the page from blocking, meaning the user can interact with the page while the video component streams in. 
```

Reply:
At this point it might be a good idea to mention loading skeletons as a fallback UI. For example, Youtube uses loading skeletons extensively.